### DISCUSSION TITLE
Make a few suggestions

### DISCUSSION THREAD

**Discussion by louis5974** [AdguardTeam/AdGuardHome#4348] [Ideas]
Hi, I'm a user of AdGuard Home (Linux) and I really like the functionality of your design.
But I would like to make a few suggestions:

1. Regarding the function of "optimistic cache", what is the specific explanation? Users can't understand.
DNS is cached to AdGuard Home and will not be refreshed until TTL time?
Or exceeds the number set by the cache, and the DNS cache will be discarded if it is not used for a long time?
Please explain to me officially, how to set TTL time and how to set "optimistic cache" when there is plenty?

2. On the "Query logs" page, currently I need to check a certain day and a certain period of time. I can only check down from the current list directory, which takes a long time. There are many "Query logs" lists in AdGuard Home, which are very unfriendly to view. I really wish the AdGuard company would develop a button to set a specific time and view the logs for a certain time. It will save us a lot of time.

**Comment by louis5974** [comment]
![Query logs](https://user-images.githubusercontent.com/53338257/156290466-a2fff322-1d53-4743-9d87-faad50afe3f9.jpg)
![optimistic cache](https://user-images.githubusercontent.com/53338257/156290469-e5db065a-f303-4be9-98b1-538becae2c3b.jpg)

**Comment by louis5974** [comment]
I would like to hear back from AdguardTeam. Also looking to enhance and improve the logging functionality.

**Comment by louis5974** [comment]


**Reply by dsheets** [reply]
I'm just a contributor -- I'm not on the AdGuard team.

**Reply by louis5974** [reply]
Sorry to bother you.

**Reply by qbit** [reply]
Also not on the team.

**Comment by ainar-g** [comment]
Please don't ping people randomly.

The “Optimistic cache” functionality is described both in the UI and on the “[Configuration][1]” page in the Wiki.  It makes AGH respond from cache even when the TTL of the cached record has already expired.

Filtering of the query log by date is planned, see #481.

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration

**Reply by louis5974** [reply]
When will query log filtering be released to the official version?
My current version is v0.107.4 . 
thank you for your reply ！
![微信截图_20220303153843](https://user-images.githubusercontent.com/53338257/156518507-21309a7e-6a8c-4346-b97c-3afeb07fe6bf.png)

**Reply by louis5974** [reply]
Query the log page, I want to be able to filter the specific date and time.  
If I need to modify the code, or program operation, I don't know how to operate. I don't quite understand about this.
